elf2table/src/lib.rs

```diff
@@ -21,10 +21,21 @@ pub use defmt_decoder::Table;
 use defmt_decoder::{StringEntry, TableEntry};
 use object::{Object, ObjectSection};
 
-/// Parses an ELF file and returns the decoded `defmt` table
+/// Parses an ELF file and returns the decoded `defmt` table.
 ///
-/// This function returns `None` if the ELF file contains no `.defmt` section
+/// This function returns `None` if the ELF file contains no `.defmt` section.
 pub fn parse(elf: &[u8]) -> Result<Option<Table>, anyhow::Error> {
+    parse_impl(elf, true)
+}
+
+/// Like `parse`, but does not verify that the defmt version in the firmware matches the host.
+///
+/// Use with caution, this is meant for defmt/probe-run development only.
```

Comment:
nitpick: I'd put this at the top of the docs so it can't be overseen. plus the `--ignore-version-mismatch`probe-run flag should get a "CAUTION: .." doc too?